File: 2015/001/part_one.py
import common

if __name__ == "__main__":
    for input_file in common.inputs:
        print(input_file)
        for line in common.read_file(input_file):
            floor = 0
            for instruction in line:
                if instruction == "(":
                    floor += 1
                elif instruction == ")":
                    floor -= 1
                else:
                    raise ValueError("Unknown instruction %s" % (instruction,))

            print(floor)

File: kotlin/src/main/kotlin/pt/dmb/adventofcode/y2015/p01/PartOne.kt
package pt.dmb.adventofcode.y2015.p01

import pt.dmb.adventofcode.common.Utilities

class PartOne {
    private var floor = 0

    fun parseInstruction(instruction: Char) {
        when (instruction) {
            '(' -> {
                floor += 1
            }

            ')' -> {
                floor -= 1
            }

            else -> {
                throw IllegalArgumentException("Unknown instruction $instruction")
            }
        }
    }

    fun floor(): Int = floor
}

fun main() {
    Utilities.readInputFile(2015, 1, scope = "main").forEach { line ->
        val solver = PartOne()
        line.toCharArray().forEach { char ->
            solver.parseInstruction(char)
        }
        println(solver.floor())
    }
}
